src/table/src/data_source.rs

```diff
@@ -14,16 +14,26 @@
 
 use std::sync::Arc;
 
+use common_query::prelude::Expr;
 use common_recordbatch::SendableRecordBatchStream;
 use store_api::storage::ScanRequest;
 
 use crate::error::Result;
+use crate::metadata::FilterPushDownType;
 
 /// This trait represents a common data source abstraction which provides an interface
 /// for retrieving data in the form of a stream of record batches.
 pub trait DataSource {
     /// Retrieves a stream of record batches based on the provided scan request.
     fn get_stream(&self, request: ScanRequest) -> Result<SendableRecordBatchStream>;
+
+    /// Tests whether the table provider can make use of any or all filter expressions
+    /// to optimise data retrieval.
+    ///
+    /// For a specific filter expression (`Expr`), if you specify `Inexact` or `Exact`,
+    /// it will include that filter expression in the `filters` field of the `ScanRequest`
+    /// that is passed to the `get_stream` function.
+    fn supports_filters_pushdown(&self, filters: &[&Expr]) -> Result<Vec<FilterPushDownType>>;
```

Comment:
`ThinTable` might be a more appropriate place for this method, as this information is usually fixed (e.g., a temporary table doesn't need to support filter pushdown, while the base table inherits this attribute from the engine)